### ISSUE TITLE
Friendly invest

### ISSUE THREAD

**Issue by Cheeeta** [mainIine/foe-helfer-extension#1018]
https://prnt.sc/sf8rzz
here :) as you can see :) is by calculator still posible to invest 3 place ... but in real is not cause for this amount that is wtited here i will not get 3rd place but 4th 
will be great if the system not show 3rd place as a friendly invest when this place is actualy invested from other man, and show 4th what is still free :) ... 3rd plce will be still visible in sniping :)

**Comment by andrgin** [comment]
I see the problem, but it is a bit more complicated as there are 2 ways to solve these situations:
a) get P4 for 57 and tell the owner to lock P4
b) get P3 for 238+72 and tell the owner to refund the 72FP

**Comment by scibuff** [comment]
I don't really see an issue here. The way I understand the calculator is that the right column shows the cost at the given Arc bonus. It knows that you can no longer get slots 1 and 2 because it is not possible to add that many FPs. But the 3rd slot can still be taken and with the bonus of 90% it correctly shows that you need to invest 238+72 to lock the slot (which means you'll lose 72 FPs).

TL;DR; the calculator still shows the 3rd slot because you can still get it (it is not locked), which I believe is the correct behavior

**Comment by Cheeeta** [comment]
andrigin:
both is wrong cause i do not sniping clan people ... and that who is 3rd is from clan too .. i dont want to delete him from his fiendly place cause he is friendly too

scibuff
calculator works right but calculator show me that FRIENDLY place is stil abailable for the same place what is actualy placed from other guy ... calculator let me gice the same price for 3nd place but if i do it i get 4th ..... if i give here 1 point more so its not friendly invest for 1,9 .... what more .. is not friendly to move guys from clan to lower places this way 
---------------------------------------
for sniping is all right ... but for friendly invest not cause this way i can make problems to me or other friendly clan investors .... and its not frendly

its not fatal bug but its isue ad the calculator will works beter for preventing unfriendlyness or refundable mistakes.

**Comment by Cheeeta** [comment]
im sorry for that thick letters but i dont know how to make them normal :D

**Comment by scibuff** [comment]
OK, I think you misunderstand what the calculator is showing ... it is not telling you how many FPs you need to add to lock the place but rather it shows how much that place earns with the set Arc bonus. In this particular case

![image](https://user-images.githubusercontent.com/722049/82057991-6a560480-96c4-11ea-956e-62c981529d6e.png)

It is showing you that "naklady" (the cost) to lock the 3rd slot is 238+72 (out of which 72 would be your loss if you did so)

**Comment by andrgin** [comment]
There are 2 things displayed:
1.) The FP that you are supposed to pay (238)
2.) The amount at which the place is safe (238+72)

It is up to the player to decide what to do now. There are several options like paying 238+72 and asking for refund, waiting until the place is locked etc.

A gray line means the place is not reachable any more or it does not make sense.
A red line means a place is still reachable, but not safe.

What makes this situation complicated is that it should be neither a gray line (because 238+72 is still possible) nor a red line (because the 238 are not possible). I will need to think of some other formatting like red background with gray text.

@scibuff No the amount of FP you earn is not shown in the calculator, only he profit. The shown values are always based on the friendly invest factor (either 85 or 90), not your own arc bonus.

**Comment by scibuff** [comment]
... @andrgin one of the friendly invest factors is always your current arc bonus. In his case, his bonus is 90% and that's what is currently shown!

**Comment by andrgin** [comment]
If friendly invest factor and your arc factor are the same, yes it shows the amount of FP you will get back, but that is not what the column shows you. It shows you "Costs for friendly investment".

**Comment by scibuff** [comment]
Not sure what you mean by "costs for *friendly* investment". Technically, the column shows how much you earn back at the specified arc bonus level (which is exactly what I wrote)

**Comment by andrgin** [comment]
Here is an example:
My arc bonus is 90%
I select friendly invest with 85% because it is a 1.85 group
Netto rewards are 600, 300, 100, 25 and 5FP
The first "costs for friendly investment" shows how much I am supposed to pay for the 1.85 group:
P3 = 100 * 1.85 = 185

The 2nd column shows how much profit I am making with this:
100*1.9 - 185 = 5FP
The amount of FP I am getting back (190) is not shown.

**Comment by scibuff** [comment]
Yes, so you specified 1.85 bonus level and it shows you how much you'd get back (if you were) at that bonus level. 

BTW, where on earth did you find a 1.85 group?! I want to join one of those :)
